src/providers/connectionStatus.tsx

```diff
@@ -1,23 +1,37 @@
-import React, { useCallback, useContext, useEffect, useRef, useState } from 'react'
+import React, { useCallback, useEffect, useRef } from 'react'
+import create from 'zustand'
 
 import { useSnackbar } from '@/providers/snackbars'
 
 export type ConnectionStatus = 'connected' | 'disconnected' | 'connecting'
 
-const ConnectionStatusContext = React.createContext<undefined | ConnectionStatusValue>(undefined)
-ConnectionStatusContext.displayName = 'ConnectionStatusContext'
-
 type ConnectionStatusValue = {
-  setNodeConnection: (connection: ConnectionStatus) => void
   nodeConnectionStatus: ConnectionStatus
   isUserConnectedToInternet: boolean
+  setNodeConnection: (connection: ConnectionStatus) => void
+  setInternetConnection: (value: boolean) => void
```

Comment:
```suggestion
  setInternetConnection: (isConnected: boolean) => void
```
last small thing - I think we should be more semantic here, let's just rename it